go/host/host.go

```diff
@@ -847,31 +847,27 @@ func (h *host) handleBatches(encodedBatches *common.EncodedBatches) error {
 		return nil
 	}
 
-	// We store the batches.
-	err = h.batchManager.StoreBatches(batches)
-	if err != nil {
-		if !errors.Is(err, batchmanager.ErrBatchesMissing) {
-			return fmt.Errorf("could not store batches. Cause: %w", err)
+	// We store the batches and submit them to the enclave.
+	for _, batch := range batches {
+		if err = h.enclaveClient.SubmitBatch(batch); err != nil {
+			return fmt.Errorf("could not submit batch. Cause: %w", err)
 		}
 
-		// We have encountered missing batches. We abort the storage operation and request the missing batches.
-		batchRequest, err := h.batchManager.CreateBatchRequest(h.config.P2PPublicAddress)
+		err = h.batchManager.StoreBatch(batch)
 		if err != nil {
-			return fmt.Errorf("could not create batch request. Cause: %w", err)
-		}
-		if err = h.p2p.RequestBatches(batchRequest); err != nil {
-			return fmt.Errorf("could not request historical batches. Cause: %w", err)
-		}
-
-		// If we requested any batches, we return early and wait for the missing batches to arrive.
-		return nil
-	}
+			if !errors.Is(err, batchmanager.ErrBatchesMissing) {
```

Comment:
Is there a risk of issues if the host storage gets out of sync with the enclave storage? 

Do the batches need to be stored on the host at all? Maybe just as a caching layer that falls back to the enclave.

Can maybe just have a `todo` for that for now but I think if the issue ever does arise it'll be a thorny one (e.g. the host requesting a batch, failing the submission to the enclave because the enclave already has it but the original persistence failed, and then requesting it again in an infinite loop)